client/state/products-list/selectors/get-product-introductory-offer.ts

```diff
@@ -0,0 +1,18 @@
+import 'calypso/state/products-list/init';
+import type { ProductIntroductoryOffer } from './get-products-list';
+import type { AppState } from 'calypso/types';
+
+/**
+ * Retrieves an introductory offer for the product with the specified slug.
+ *
+ * @param {object} state - global state tree
+ * @param {string} productSlug - internal product slug, eg 'jetpack_premium'
+ * @returns {import('./get-products-list').ProductIntroductoryOffer|null} - introductory offer details for the product, or null if no offer exists
+ */
+
```

Comment:
Does this docblock makes sense on a TS? The function definition seems to me already self descriptive enough. Plus also I think there is an extra space that we usually don't include between the docblock and the function.

Reply:
I was mostly replicating the approach for `getProductBySlug()`/`get-product-by-slug.ts` in the same directory.

I'll remove the extra whitespace.